python/ray/util/client/server/dataservicer.py

```diff
@@ -19,6 +20,20 @@
 logger = logging.getLogger(__name__)
 
 
+def fill_queue(grpc_input_generator, output_queue) -> None:
+    """
+    Pushes incoming requests to a shared output_queue.
+    """
+    try:
+        for req in grpc_input_generator:
+            output_queue.put(req)
+    except grpc.RpcError as e:
+        logger.debug("closing dataservicer reader thread "
+                     f"grpc error reading request_iterator: {e}")
+    finally:
+        output_queue.put(None)
```

Comment:
why is this necessary?

Reply:
Because the iterator over the `output_queue` stops iterating when it sees a `None` value.